primefaces/src/main/java/org/primefaces/component/api/UIData.java

```diff
@@ -557,7 +536,41 @@ else if (component instanceof UIForm) {
                 saveDescendantState(facet, context);
             }
         }
+    }
+
+    protected void saveInputState(EditableValueHolder input, FacesContext context) {
+        String componentClientId = ((UIComponent) input).getClientId(context);
+        if (isDefaultAndEmpty(input)) {
+            getStateHelper().remove(PropertyKeys.saved, componentClientId);
```

Comment:
this code looks weird?
It actually returns a Map with Map<String, SavedState> for ALL inputs, should we really remove the whole map for the single componentClientId?